blocks/kanban-board/src/components/board/column/column.tsx

```diff
@@ -0,0 +1,109 @@
+import { SortableContext, useSortable } from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import clsx from "clsx";
+
+import { IconButton } from "../../icon-button/icon-button";
+import { DiscardIcon } from "../../icons/discard-icon";
+import { PlusIcon } from "../../icons/plus-icon";
+import { Card } from "../card/card";
+import {
+  ColumnData,
+  CreateCardCallback,
+  DeleteCardCallback,
+  DeleteColumnCallback,
+  UpdateCardContentCallback,
+  UpdateColumnTitleCallback,
+} from "../types";
+import { EditableColumnTitle } from "./editable-column-title/editable-column-title";
+import styles from "./styles.module.scss";
+
+interface ColumnProps {
+  data: ColumnData;
+  deleteColumn: DeleteColumnCallback;
+  createCard: CreateCardCallback;
+  deleteCard: DeleteCardCallback;
+  updateColumnTitle: UpdateColumnTitleCallback;
+  updateCardContent: UpdateCardContentCallback;
+  readonly?: boolean;
+}
+
+export const Column = ({
+  data,
+  deleteColumn,
+  createCard,
+  deleteCard,
+  updateCardContent,
+  updateColumnTitle,
+  readonly,
+}: ColumnProps) => {
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+    isDragging,
+    isSorting,
+  } = useSortable({ id: data.id });
+
+  const style = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+    opacity: isDragging ? 0.5 : 1,
+  };
+
+  return (
+    <div
+      className={styles.wrapper}
+      ref={setNodeRef}
+      style={style}
+      {...attributes}
+    >
+      <div
```

Comment:
Would it be possible to make this only a wrapper component for the drag and drop logic? That way `StaticColumn` could be a pure component and this could wrap it

Reply:
Hmm, `attributes` and `listeners` are spreaded into different elements, wouldn't that cause a problem?

Reply:
Just the 2 outside divs have those right? You can extract what's inside

Reply:
Updated here https://github.com/hashintel/hash/pull/2234/commits/c5cdbc91eee25910996f76c1ae702c25f34305b4